src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -269,21 +306,44 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
                 $requestBody = new RequestBody(sprintf('The %s %s resource', HttpOperation::METHOD_POST === $method ? 'new' : 'updated', $resourceShortName), $this->buildContent($requestMimeTypes, $operationInputSchemas), true);
             }
 
-            $pathItem = $pathItem->{'with'.ucfirst($method)}(new Model\Operation(
-                $operationId,
-                $operation->getOpenapiContext()['tags'] ?? [$operation->getShortName() ?: $resourceShortName],
-                $responses,
-                $operation->getOpenapiContext()['summary'] ?? $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface),
-                $operation->getOpenapiContext()['description'] ?? $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface),
-                isset($operation->getOpenapiContext()['externalDocs']) ? new ExternalDocumentation($operation->getOpenapiContext()['externalDocs']['description'] ?? null, $operation->getOpenapiContext()['externalDocs']['url']) : null,
-                $parameters,
-                $requestBody,
-                isset($operation->getOpenapiContext()['callbacks']) ? new \ArrayObject($operation->getOpenapiContext()['callbacks']) : null,
-                $operation->getOpenapiContext()['deprecated'] ?? (bool) $operation->getDeprecationReason(),
-                $operation->getOpenapiContext()['security'] ?? null,
-                $operation->getOpenapiContext()['servers'] ?? null,
-                array_filter($operation->getOpenapiContext() ?? [], static fn ($item): int|false => preg_match('/^x-.*$/i', (string) $item), \ARRAY_FILTER_USE_KEY)
-            ));
+            $openapiOperation = ($hasOpenapi ? $openapiOperation : new Model\Operation())
+                // Defaults
+                ->withTags($hasOpenapi && $openapiOperation->getTags() ? $openapiOperation->getTags() : [$operation->getShortName() ?: $resourceShortName])
+                ->withSummary($hasOpenapi && $openapiOperation->getSummary() ? $openapiOperation->getSummary() : $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface))
+                ->withDescription($hasOpenapi && $openapiOperation->getDescription() ? $openapiOperation->getDescription() : $this->getPathDescription($resourceShortName, $method, $operation instanceof CollectionOperationInterface))
+                ->withDeprecated($hasOpenapi && $openapiOperation->getDeprecated() ? $openapiOperation->getDeprecated() : (bool) $operation->getDeprecationReason());
```

Comment:
I'm then not sure about this section, either we try to have sane defaults in the `new Model\Operation` (from the suggestion above). Or we do:

```
if (null === $openapiOperation->getTags()) {
  $openApiOperation->withTags([...]);
}

etc.
```

This means that we should fix our getters in Openapi\Model (which isn't perfect as if we want to reflect a property from the specification as **required** I think it's correct).